File: DSA/SearchAlgorithms/LinearSearch.py
#linear search algorithm use the sequential approch to find the targeted data.
# Best-case complexity is O(1)
# Worst-case Performance is O(n)
# Average Performance O(n/2)

def LinearSearch(arr,target):

    for i in range(len(arr)):
        if(arr[i]==target):
            return i


    return -1


lst = [1,2,3,4,5,6,7,8,9,10]

target = int(input("value to be found : "))

result = LinearSearch(lst,target)

if(result == -1):
    print("target element is not found.")

else:
    print("element is present at the index ",result)

File: DSA/SearchAlgorithms/LinearSearch.cpp
#include<iostream>

using namespace std;

int LinearSearch(int arr[],int value,int length)
{
	
	for(int i=0;i<length;i++)
	{
		if(arr[i]==value)
		{
			cout<<"value found at index ";
			return i;
		}

		
	}
	cout<<"target value i not in the data.";
	return -1;
}

int main()
{
	int arr[]={1,2,3,4,5,6,7,8,9,0};
	int target;
	cout<<"Enter the value to be found ";
	cin>>target;
	int l=sizeof(arr)/sizeof(int);
	cout<<LinearSearch(arr,target,l)<<endl;
}
